volatility3/framework/plugins/linux/pslist.py

```diff
@@ -57,7 +78,37 @@ def _generator(self):
             if task.parent:
                 ppid = task.parent.pid
             name = utility.array_to_string(task.comm)
-            yield (0, (pid, ppid, name))
+            uid = task.real_cred.uid.val
+            gid = task.real_cred.gid.val
+
+            # task.(real_)start_time contains amount of nanoseconds since boottime
+            create_timespec = task.real_start_time if hasattr(task, "real_start_time") else task.start_time
+            create_timespec_sec = int(create_timespec / 1e9)
+            create_timespec_nsec = create_timespec % 1e9
+            create_time = datetime.datetime.fromtimestamp(boot_time_nano / 1e9) + datetime.timedelta(
+                seconds = create_timespec_sec, microseconds = int(create_timespec_nsec / 1e3))
+
+            file_output = "Disabled"
+            if self.config['dump']:
+                proc_layer_name = task.add_process_layer()
+                if not proc_layer_name:
+                    continue
+
+                # Find the vma that belongs to the main ELF of the process
+                vma = None
+                for v in task.mm.get_mmap_iter():
+                    if v.vm_start == task.mm.start_code:
+                        vma = v
+                        break
+                if vma is not None:
+                    file_handle = elfs.Elfs.elf_dump(self.context, proc_layer_name, elf_table_name, vma, task,
+                                                     self.open)
+                file_output = "Error outputting file"
+                if file_handle:
+                    file_handle.close()
+                    file_output = str(file_handle.preferred_filename)
+
```

Comment:
The logic in lines 99-109 seems complicated, and I am fairly certain the plugin would backtrace if 'vma' is None going into line 103 as the file_handle variable would never be declared before 107 is hit.

It seems something like this would achieve the same goals (break after the right VMA is found, write out the file, report an error, etc.) with less complicated code flow.

```
file_output = "Error outputting file"

for v in task.mm.get_mmap_iter():
  if v.vm_start == task.mm.start_code:
       file_handle = elfs.Elfs.elf_dump(self.context, .....)
       if file_handle:
             file_output = str(file_handle.preferred_filename)
             file_handle.close()
             
       break
```